src/server/index.js

```diff
@@ -20,6 +20,17 @@ if (!process.env.NEXTAUTH_URL) {
   logger.warn("NEXTAUTH_URL", "NEXTAUTH_URL environment variable not set")
 }
 
+if (
+  process.env.NODE_ENV === "development" &&
+  !/^http:\/\/localhost:\d+$/.test(process.env.NEXTAUTH_URL) &&
+  !process.env.NEXT_PUBLIC_NEXTAUTH_URL
+) {
+  logger.warn(
+    "NEXT_PUBLIC_NEXTAUTH_URL",
+    `NEXTAUTH_URL is set to "${process.env.NEXTAUTH_URL}" instead of the default "http://localhost:\${PORT}", and NEXT_PUBLIC_NEXTAUTH_URL is not set. Client side path detections will fail.`
+  )
+}
```

Comment:
In my opinion, we should move this logic to `parseUrl` and the check above too to start sliming down this monolith file. 

On the other hand, we should be throwing errors in both cases rather than warnings as this misconfiguration, I understand, is critical? 🤔

Reply:
> On the other hand, we should be throwing errors in both cases rather than warnings as this misconfiguration, I understand, is critical? 

I agree on this. I could not think of a better message, so I left it as warning, but it should definitely be treated as error.